File: 1002-find-common-characters/1002-find-common-characters.py
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        if not words:
            return []
        hashmap = [0] * 26
        ans = []
        for c in words[0]:
            hashmap[ord(c) - ord("a")] += 1
        for i in range(1, len(words)):
            hashOther = [0] * 26
            for c in words[i]:
                hashOther[ord(c) - ord("a")] += 1
            for j in range(26):
                hashmap[j] = min(hashmap[j], hashOther[j])
        
        for i in range(26):
            while hashmap[i]:
                ans.append(chr(i + ord("a")))
                hashmap[i] -= 1
        return ans
            
        

File: 1002-find-common-characters/1002-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        vector<string> result;
        if (words.size() == 0) return result;
        int record[26] = {0};
        for (int i = 0; i < words[0].size(); i++){
            record[words[0][i] - 'a']++;
        }
        
        int recordOther[26] = {0};
        for (int i = 1; i < words.size(); i++){
            memset(recordOther, 0, 26 * sizeof(int));
            for (int j = 0; j < words[i].size(); j++){
                recordOther[words[i][j] - 'a']++;
            }
            for (int k = 0; k < 26; k++){
                record[k] = min(record[k], recordOther[k]);
            }
        }
        
        for (int i = 0; i < 26; i++){
            while (record[i] != 0){
                string s(1, i + 'a');
                result.push_back(s);
                record[i]--;
            }
        }
        return result;
    }
};